src/ceph-disk/ceph_disk/main.py

```diff
@@ -2545,6 +2567,60 @@ def get_dmcrypt_type(args):
             return None
 
 
+class Secrets(object):
+
+    def __init__(self):
+        secret, stderr, ret = command(['ceph-authtool', '--gen-print-key'])
+        LOG.debug("stderr " + stderr)
+        assert ret == 0
```

Comment:
command_check_call does not capture the output and cannot be used here. An assert is used here because this should never happen except if the installation is corrupted (i.e. ceph-authtool has not been installed or various other "end of the world" conditions). Issuing a sensible error message to the user at that point is not only pointless, it is also most likely very far away from the real reason why the error happens. So we assert and do not go any further.